File: tutorials/30_Days_Of_Code/python/d14_Scope.py
class Difference:
    maximumDifference = 0

    def __init__(self, a):
        self.__elements = a

    # Add your code here
    def computeDifference(self):
        maxno = max(self.__elements)
        minno = min(self.__elements)
        self.maximumDifference = abs(minno - maxno)

# End of Difference class

_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)

_ = input()
a = [int(e) for e in input().split(' ')]

e = Difference(a)
e.computeDifference()
print(e.maximumDifference)

File: tutorials/30_Days_Of_Code/cpp/d14_scope.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference {
    private:
    vector<int> elements;
  
  	public:
  	int maximumDifference;

    Difference(vector<int> &a) {
        this->elements = a;
    }

    void computeDifference() {
        int smallest_element = *min_element(elements.begin(), elements.end());
        int largest_element  = *max_element(elements.begin(), elements.end());
        maximumDifference = abs(largest_element - smallest_element);
    }
};

int main() {
    int N;
    cin >> N;
    
    vector<int> a;
    
    for (int i = 0; i < N; i++) {
        int e;
        cin >> e;
        
        a.push_back(e);
    }

    Difference d(a);

    d.computeDifference();

    cout << d.maximumDifference;

    return 0;
}
